extension/dynamicconfigextension/config.go

```diff
@@ -0,0 +1,33 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dynamicconfigextension
+
+import (
+	"go.opentelemetry.io/collector/config/configmodels"
+)
+
+// Config has the configuration for the extension enabling the dynamic
+// configuration service
+type Config struct {
+	configmodels.ExtensionSettings `mapstructure:",squash"`
+
+	// Endpoint is the address and port used to communicate the config updates
+	// The default value is localhost:55700.
+	Endpoint string `mapstructure:"endpoint"`
```

Comment:
Since we allow specifying config in a file, this Endpoint is optional right? Then, how do we use its default value?

I think what we should think about all cases and handle them properly:

(1) no Endpoint; no LocalConfigFile: should we report error or use a fixed schedule for all metric? I prefer reporting error because dynamicconfig becomes meanless;

(2) has Endpoint; no LocalConfigFile: obviously, we use Endpoint, but what if Endpoint is not available? Reporting error?

(3) no Endpoint; has LocalConfigFile: obviously, we use LocalConfigFile.

(4) has Endpoint; has LocalConfigFile: prefer Endpoint, but fall back to LocalConfigFile?

For now, I think it is okay to simplify this and we only allow (2) and (3). That is, use either Endpoint  or  LocalConfigFile, if it does not work, report error.

Reply:
Great points, both Endpoint and LocalConfigFile are optional, with default values provided in `factory.go`

Endpoint refers to the endpoint that an agent/sdk communicates with to receive updated metrics. The naming is a little ambiguous in this case, so I can rename it to something like AgentEndpoint?

Eventually, to communicate with third party backends and upstream collectors, I will add additional configs, maybe something like "UpstreamEndpoint" (?), which would specify the address to retrieve configs from third-party backends or upstream collectors.

The LocalConfigFile defaults to a file in the same directory as the collector configs, named `dynamic-config-local-schedules.yaml`. If the file is not found, or if a socket on Endpoint cannot be opened, then errors are propogated upwards and the program will terminate.